rasa/core/test.py

```diff
@@ -105,26 +104,104 @@ def has_prediction_target_mismatch(self) -> bool:
 
     def serialise(self) -> Tuple[List[Text], List[Text]]:
```

Comment:
This method is now very long. Can you please split it up? We could extract the entity prediction calculation into a separate method.